### PR TITLE
QEMU UEFI rework

### PR DESCRIPTION
**Pull request by gctucker**
This depends on #160 to be merged first, and is a clean-up of #161.

### CODE REVIEW

File: test-configs.yaml [hunk 1]
```diff
@@ -751,19 +735,55 @@ device_types:
       - whitelist: {defconfig: ['defconfig']}
       - blacklist: {kernel: ['v3.']}
 
-  qemu_arm:
+  qemu_arm-gicv2:
```
Comment (montjoie) [inline test-configs.yaml]: I prefer to qualify better the machine, it is a virt-gicv2

Comment (gctucker) [reply test-configs.yaml]: Sure.

**Comment by mgalka** [comment]
I'm not entirely convinced with the "base_name" attribute. It's certainly better than a "type_name", maybe we could use something like "type_" ?
Apart from that LGTM.

**Comment by gctucker** [comment]
> maybe we could use something like "type_" ?

How would you intuitively make the difference between `DeviceType.name` and `DeviceType.type_`?  That's essentially what we had before with `.type_name` and it was very vague.

The point of `.base_name` is to show that it's a "lower-level" name, something other device types are derived from in the YAML definitions.  Also we can have something similar for `TestPlan.base_name`, there's another PR coming for that.

**Comment by mgalka** [comment]
> How would you intuitively make the difference between `DeviceType.name` and `DeviceType.type_`? That's essentially what we had before with `.type_name` and it was very vague.

Well, it somehow looked to me as a class-metaclass relationship in Python, hence "type_" seemed valid for me, but what seems clear to me doesn't necessarily be a good solution.

> The point of `.base_name` is to show that it's a "lower-level" name, something other device types are derived from in the YAML definitions. Also we can have something similar for `TestPlan.base_name`, there's another PR coming for that.

Got your point. The consistent naming looks good.

**Comment by gctucker** [comment]
Tested OK on staging:
https://staging.kernelci.org/boot/all/job/kernelci/branch/staging.kernelci.org/kernel/staging-20190829.3/
https://staging-storage.kernelci.org/kernelci/staging.kernelci.org/staging-20190829.3/arm/multi_v7_defconfig/gcc-8/lab-baylibre-dev/boot-qemu_arm-virt-gicv2.html

Also got the UEFI jobs running as expected in the LKFT lab (7 variants, here are a couple):
https://lkft.validation.linaro.org/scheduler/alljobs?length=25&search=uefi#table
https://lkft.validation.linaro.org/scheduler/job/893837
https://lkft.validation.linaro.org/scheduler/job/893834

**Pull request merged**